briefcase/app.py

```diff
@@ -219,14 +222,32 @@ def generate_app_template(self, extra_context=None):
             extra_context=_extra_context
         )
 
-    def git_checkout(self, path):
+    def _has_cookiecutter_json(self, template_path):
+        cookiecutter_json_path = os.path.expanduser('%s/cookiecutter.json' % template_path)
+        return os.path.exists(cookiecutter_json_path)
+
+    def _get_all_branches(self, path):
+        branches = subprocess.check_output(["git", "ls-remote", "--heads"], stderr=subprocess.STDOUT, cwd=path)
+        branches = str(branches).split("\\n")
```

Comment:
This should probably decode the git output with something like `branches.decode('utf-8').splitlines()`.